dht.go

```diff
@@ -144,11 +144,13 @@ func NewDHTClient(ctx context.Context, h host.Host, dstore ds.Batching) *IpfsDHT
 }
 
 func makeDHT(ctx context.Context, h host.Host, dstore ds.Batching, protocols []protocol.ID, bucketSize int) *IpfsDHT {
-	rt := kb.NewRoutingTable(bucketSize, kb.ConvertPeerID(h.ID()), time.Minute, h.Peerstore())
+	self := kb.ConvertPeerID(h.ID())
+	rt := kb.NewRoutingTable(bucketSize, self, time.Minute, h.Peerstore())
 	cmgr := h.ConnManager()
 
 	rt.PeerAdded = func(p peer.ID) {
-		cmgr.TagPeer(p, "kbucket", 5)
+		dist := kb.CommonPrefixLen(self, kb.ConvertPeerID(p))
+		cmgr.TagPeer(p, "kbucket", 5+dist)
 	}
 
 	rt.PeerRemoved = func(p peer.ID) {
```

Comment:
for completeness purposes, we should be erasing the tag on peer removal.

Reply:
We are, right below this?